Formula/bitcoin.rb

```diff
@@ -23,9 +23,12 @@ class Bitcoin < Formula
 
   depends_on "autoconf" => :build
   depends_on "automake" => :build
+  # berkeley db should be kept at version 4
+  # https://github.com/bitcoin/bitcoin/blob/master/doc/build-osx.md
+  # https://github.com/bitcoin/bitcoin/blob/master/doc/build-unix.md
+  depends_on "berkeley-db@4"
```

Comment:
Move this below `depends_on "pkg-config" => :build` and above `depends_on "boost"`.

Reply:
Ah, sorry, I didn't realize it was build dependencies first.

Reply:
Sure. No problem. Can we add a test that fails when `berkeley-db@4` is changed to a newer version?

Reply:
I think that's a little bit beyond me, I don't actually know Ruby.

The test would have to:
- Create a tmpdir
- Spawn bitcoind in regtest mode with the tempdir as the data dir. Additionally, you'd have to make sure that the node was configured with ports that wouldn't conflict with a bitcoind instance that the user is running in regtest mode
- Create a legacy wallet (if it fails, bdb wasn't available, or was the wrong version during buidl)
- Delete the tempdir
- Kill the spawned bitcoind instance

Reply:
> - Spawn bitcoind in regtest mode with the tempdir as the data dir. Additionally, you'd have to make sure that the node was configured with ports that wouldn't conflict with a bitcoind instance that the user is running in regtest mode
> 
> - Create a legacy wallet (if it fails, bdb wasn't available, or was the wrong version during buidl)

Can you outline what commands you would run in your shell to do these steps? The other three are handled by `brew` mostly automatically.

Reply:
Sure, something like this:

```
TMPDIR=`mktemp -d`
RPCPORT=`freeport`
bitcoind -regtest -rpcport=$RPCPORT -listen=0 -datadir=$TMPDIR &
bitcoin-cli -regtest -datadir=$TMPDIR -rpcport=$RPCPORT createwallet test-wallet false false '' false false
# the above bitcoin-cli command will fail if RPC wallets aren't supported
# kill bitcoind launched above
```

Reply:
Ok, I've pushed a change that I think should do that. Please try it out locally to see if it works too. Presumably it should fail with the existing version of `bitcoind`.

Reply:
Using the test with `berkeley-db@5` seems to error with `error: Error parsing JSON: ''`. Not sure if that's the error we're looking for?

Reply:
Oh, I see. I left out an argument...

Reply:
I believe the argument that's currently `"''"` should just be `""`

Reply:
Great, this test seems to work. When built against `berkeley-db@5`, we get:
```
error code: -4
error message:
Compiled without bdb support (required for legacy wallets)
```
https://github.com/carlocab/workflow-test/runs/7946821335?check_suite_focus=true#step:5:251